lib/core/connector_job.rb

```diff
@@ -133,5 +148,24 @@ def self.fetch_jobs_by_query(query, page_size)
 
       results
     end
+
+    def initialize(es_response)
+      # TODO: remove the usage of with_indifferent_access. The initialize method should expect a hash argument
+      @elasticsearch_response = es_response.with_indifferent_access
+    end
+
+    def terminate!(status, error = nil, ingestion_stats = {}, connector_metadata = {})
+      ingestion_stats ||= {}
+      ingestion_stats[:total_document_count] = ElasticConnectorActions.document_count(index_name)
```

Comment:
This part looks a bit scary - it should already come from ingestion stats, right?

Reply:
No, ingestion stats only report how many docs are ingested by this job. This is how many docs are in the connector at the end of the job, requested by @sphilipse . And that's why it's only updated when `terminate!` is called (i.e. terminal status like `complete`, `canceled` or `error`)